### PR TITLE
Add iterator traits aliases to ConstPointerContainer::ConstIterator

### PR DESCRIPTION
**Pull request by skottmckay**

### CODE REVIEW

**Comment by edgchen1** [comment]
Thanks for addressing this.

There are a few other input iterator things I'm not sure we are handling (from here: https://en.cppreference.com/w/cpp/named_req/InputIterator):
- pre-increment operator returns void instead of ConstIterator&
- no post-increment operator
- no operator->()

File: include/onnxruntime/core/common/const_pointer_container.h [hunk 1]
```diff
@@ -20,17 +20,41 @@ class ConstPointerContainer {
...
+    const T*& operator*() const {
+      item_ = *current_;
+      return item_;
+    }
+
+    const T** operator->() const { return &(operator*()); };
 
    private:
     const_iterator current_;
+    mutable const T* item_;
```
Comment (edgchen1) [inline include/onnxruntime/core/common/const_pointer_container.h]: Why was `item_` introduced? Is it possible to use `current_` instead?

Comment (skottmckay) [reply include/onnxruntime/core/common/const_pointer_container.h]: To satisfy returning T*& to match the usual iterator traits for the 'reference' type. Have to have something local that we can return a reference to.

File: include/onnxruntime/core/common/const_pointer_container.h [hunk 2]
```diff
@@ -20,17 +20,41 @@ class ConstPointerContainer {
   class ConstIterator {
    public:
     using const_iterator = typename Container::const_iterator;
+    using iterator_category = std::input_iterator_tag;
+    using value_type = T*;
```
Comment (edgchen1) [inline include/onnxruntime/core/common/const_pointer_container.h]: not sure myself here - should value_type = const T*? and pointer = const T**, reference = const T*&?

Comment (skottmckay) [reply include/onnxruntime/core/common/const_pointer_container.h]: Matches onnx/graph_node_list.h (and we are enforcing the addition of the 'const'), so I believe it's correct. .

```
struct iterator_traits<ONNX_NAMESPACE::generic_graph_node_list_iterator<T>> {
  using difference_type = int64_t;
  using value_type = T*;
  using pointer = T**;
  using reference = T*&;
  using iterator_category = bidirectional_iterator_tag;
};
```

**Pull request merged**